engine/src/main/java/org/terasology/rendering/nui/UIWidget.java

```diff
@@ -69,6 +70,15 @@
      */
     <T extends UIWidget> T find(String id, Class<T> type);
 
+    /**
+     * Try to find a widget with the given id and type, within the current widget and its contents.
```

Comment:
Could you include a note on the PR for why this new method is included and how it differs from the regular find? Any particular reasoning / justification etc? Might help as a comment on the PR itself rather than a line comment here. Not quite sure if this would be an API issue, might be avoided since AbstractWidget is also updated.

Reply:
About this method, we have find() method which could return null and without check we could get NPE. Unfortunately, we are not always do this check. I decided to add this method because it might be useful to be sure that our code won't break anything if UI was corrupted somehow.